roachprod: update host/port flags

RaduBerinde opened the pull request:
The `--host` and `--port` flags are deprecated. This change switches
`roachprod` to using `listen-addr` and `http-addr` instead.

We also simplify the logic around `advertise-addr`: we can now access
the PrivateIP directly instead of detecting it from inside the start
script.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/73031)
<!-- Reviewable:end -->

RaduBerinde commented:
Adding do-not-merge label to remind me to run a multi-cloud test before merging.

tbg reviewed (approved):
Thank you, much appreciated.

> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/73031#-:-Mp64eui0_k9BL-SY72f:bxrhi0t)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @healthy-pod and @rail)



<!-- Sent from Reviewable.io -->

rail reviewed (approved):
Looks much cleaner! Thank you!

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/73031#-:-Mp74q1731RJd2Q7ir2N:ba4vo0b)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @healthy-pod)



<!-- Sent from Reviewable.io -->

RaduBerinde committed:
roachprod: update host/port flags

The `--host` and `--port` flags are deprecated. This change switches
`roachprod` to using `listen-addr` and `http-addr` instead.

We also simplify the logic around `advertise-addr`: we can now access
the PrivateIP directly instead of detecting it from inside the start
script.

Release note: None

RaduBerinde commented:
Well, looks like multi-cloud hasn't been working for a long time so I can't really test it. I did test that on a single non-local cluster we advertise the same kind of IP address that we did before (e.g. `10.12.*.*` for AWS).

RaduBerinde commented:
bors r+
